src/controllers/productController.js

```diff
@@ -26,48 +26,47 @@ class productController{
     }
 
   static async createProduct(req, res) {
-    try {
-
-      if (req.user.role.roleName !== "vendor") {
-        return res
-          .status(401)
-          .json({ status: "fail", message: req.t("Unauthorized") });
-      };
-
-     const {
-        productName,
-        productPrice,
-        quantity,
-        productDescription,
-        productOwner,
-        CategoryId,
-        expiredDate
-      } = req.body;
-      const productImage = req.files.map((img) => img.path);
-      const product = await Product.create({
-        vendorId: req.user.role.id,
-        productName,
-        productPrice,
-        quantity,
-        productDescription,
-        productOwner,
-        expiredDate,
-        CategoryId,
-        productImage
-      });
-
-      await product.save();
-      return res
-        .status(200)
-        .json({
+    if (req.user.roleName === "vendor") {
+      try {
+        const {
+          productName,
+          productPrice,
+          quantity,
+          productDescription,
+          productOwner,
+          CategoryId,
+          bonus,
+          expiredDate
+        } = req.body;
+        const productImage = req.files.map((img) => img.path);
+        const product = await Product.create({
+          vendorId: req.user.id,
+          productName,
+          productPrice,
+          quantity,
+          productDescription,
+          productOwner,
+          expiredDate,
+          CategoryId,
+          bonus,
+          productImage
+        });
+        await product.save();
+        return res.status(200).json({
           status: "success",
           message: "product created",
           product: { product }
         });
-    } catch (error) {
-      return res.json({ error: error.message });
+      } catch (error) {
+        return res.json({ error: error.message });
```

Comment:
- follow the same standard json response here
- also remember to put the status code e.g 

`return res.status(400).json({ status:"fail", error: error.message });